### PR TITLE
API/Initial Rest Client

### PR DESCRIPTION
**Pull request by timotiusnc**
1. Add dependency to Retrofit (+OKHttp), RxJava, and RxAndroid.
2. Add `RestClient` component to wrap API call.
3. Small PoC: requesting 20 top stories and displaying it.

### CODE REVIEW

File: app/src/main/java/io/pumpkinz/pumpkinreader/MainActivity.java [hunk 1]
```diff
@@ -8,9 +8,7 @@
 import android.support.v7.widget.Toolbar;
 import android.view.MenuItem;
 
-
 public class MainActivity extends AppCompatActivity {
-
```
Comment (wiseodd) [inline app/src/main/java/io/pumpkinz/pumpkinreader/MainActivity.java:13]: I prefer to have blank line after class declaration & before class closing bracket.

Also it's better to have double blank lines between import and class declaration. It's just easier for the eyes.

File: app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java [hunk 2]
```diff
@@ -4,56 +4,99 @@
...
+    }
+
+    @Override
     public void onViewCreated(View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
 
-        this.newsList = (RecyclerView) view.findViewById(R.id.news_list);
-        this.newsList.setHasFixedSize(true);
+        newsList = (RecyclerView) view.findViewById(R.id.news_list);
+        newsList.setHasFixedSize(true);
```
Comment (wiseodd) [inline app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java]: This should be removed, as our data will be dynamic.

File: app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java [hunk 3]
```diff
@@ -4,56 +4,99 @@
...
-                "As We May Think (1945)",
-                "Running Lisp in Production",
-                "How side projects saved our startup",
-                "Ask HN: How big does an open-source project need to be for a lifestyle business?",
-                "Fighting spam with Haskell"
-        );
-        this.newsList.setAdapter(new NewsAdapter(dataset));
+        newsList.setLayoutManager(layoutManager);
+
+        dataset = new ArrayList<>(0);
```
Comment (wiseodd) [inline app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java]: Why do you need (0)?

File: app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java [hunk 4]
```diff
@@ -4,56 +4,99 @@
...
+
+            @Override
+            public void onError(Throwable e) {
+                Log.d("stories err", e.toString());
+            }
+
+            @Override
+            public void onNext(List<ItemPOJO> itemPOJOs) {
+                Log.d("stories", String.valueOf(itemPOJOs.size()));
+                for (ItemPOJO item : itemPOJOs) {
```
Comment (wiseodd) [inline app/src/main/java/io/pumpkinz/pumpkinreader/NewsListFragment.java]: Use List.addAll();

File: app/src/main/java/io/pumpkinz/pumpkinreader/model/ItemPOJO.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+ * A Hacker News item can be one of the following (https://github.com/HackerNews/API):
+ * 1. Story
+ * 2. Comment
+ * 3. Job
+ * 4. Poll
+ * 5. Pollopt
+ */
+
+public class ItemPOJO {
+    // Common fields
```
Comment (wiseodd) [inline app/src/main/java/io/pumpkinz/pumpkinreader/model/ItemPOJO.java:21]: Please add blank line after and before opening and closing bracket respectively

File: app/src/main/java/io/pumpkinz/pumpkinreader/rest/ApiService.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package io.pumpkinz.pumpkinreader.rest;
```
Comment (wiseodd) [inline app/src/main/java/io/pumpkinz/pumpkinreader/rest/ApiService.java]: I think the package name is too specific. What if, we decided to make this realtime, and we don't use REST anymore.

I thing io.pumpkinz.pumpkinreader.service is more fitting.

**Comment by timotiusnc** [comment]
Updated!

**Pull request merged**